core/trie/trie.go

```diff
@@ -145,14 +125,85 @@ func (sn *StorageNode) Key() *Key {
 	return sn.key
 }
 
-func (sn *StorageNode) Node() *Node {
-	return sn.node
+func (sn *StorageNode) Value() *felt.Felt {
+	return sn.node.Value
+}
+
+func (sn *StorageNode) String() string {
+	return fmt.Sprintf("StorageNode{key: %s, node: %s}", sn.key, sn.node)
+}
+
+func (sn *StorageNode) Update(other *StorageNode) error {
```

Comment:
Similar to another comment, can key be (incorrectly) updated but the node not?

Reply:
Fixed.